cpp/src/arrow/testing/gtest_util.h

```diff
@@ -354,6 +354,13 @@ ARROW_TESTING_EXPORT
 std::shared_ptr<Table> TableFromJSON(const std::shared_ptr<Schema>&,
                                      const std::vector<std::string>& json);
 
+ARROW_TESTING_EXPORT
+std::shared_ptr<Tensor> TensorFromJSON(const std::shared_ptr<DataType>& type,
+                                       const std::vector<int64_t>& shape,
+                                       std::string_view json,
+                                       const std::vector<int64_t>& strides = {},
+                                       const std::vector<std::string>& dim_names = {});
```

Comment:
How about allowing `shape`, `dim_names` and `strides` be input as JSON as well?
```suggestion
ARROW_TESTING_EXPORT
std::shared_ptr<Tensor> TensorFromJSON(const std::shared_ptr<DataType>& type,
                                       std::string_view data,
                                       std::string_view shape,
                                       std::string_view dim_names = "[]",
                                       std::string_view strides = "[]");
```

Small nit, make argument order equal to `Tensor::Make`:
```cpp
  static inline Result<std::shared_ptr<Tensor>> Make(
      const std::shared_ptr<DataType>& type, const std::shared_ptr<Buffer>& data,
      const std::vector<int64_t>& shape, const std::vector<int64_t>& strides = {},
      const std::vector<std::string>& dim_names = {}) {
```

Reply:
> How about allowing `shape`, `dim_names` and `strides` be input as JSON as well?

Great idea!

> Small nit, make argument order equal to `Tensor::Make`:

Agree, thank you for the suggestion!

Reply:
@rok, do you maybe have any suggestions how to parse JSON into a  `const std::vector<...>`? I have been researching this topic a bit and found nothing I could work with.

Reply:
That's nano-neat Dewey! I did find the parse method in _nlohmann/json_ but didn't want to use it as I am trying to stick to the methods we already use in Arrow C++, if possible (yeah, no access to it). Not sure if I can solve it like that though.

Reply:
> You could also use `ArrayFromJSON()` and convert the result into a vector using `array->data()->values()`.

Aha! Was hoping this was possible but wasn't sure. I like this approach, thank you!

Reply:
We use `arrow/json/rapidjson_defs.h` for deserializing `FixedShapeTensor` - [see here](https://github.com/apache/arrow/blob/0ce72675f4dd755b2696eb6597850b21df647bb8/cpp/src/arrow/extension/fixed_shape_tensor.cc#L168-L177). But I'd use `ArrayFromJSON` if it this if possible. I assume this helper will only be used for testing?

Reply:
> I assume this helper will only be used for testing?

Yes, correct.

I decided though to use `arrow/json/rapidjson_defs.h`

I also added `TensorFromJSON` with two different signatures. I would still like to use strides as `std::vector<int64_t>` and not as `std::string_view` in the tests for `RecordBatch::ToTensor` because I am not sure if hardcoding the value of the size of tensor data type in bytes would be a good idea compared to using `sizeof()` - in case there is a difference for other compliers/architectures. It is quite possible I am complicating this =)